modAionImpl/src/org/aion/zero/impl/sync/msg/ResStatus.java

```diff
@@ -73,18 +119,46 @@ public static ResStatus decode(final byte[] _bytes) {
         bb.get(_totalDifficulty);
         bb.get(_bestHash);
         bb.get(_genesisHash);
-        return new ResStatus(_bestBlockNumber, _totalDifficulty, _bestHash, _genesisHash);
+
+        int _len = 8 + 1 + _totalDifficultyLen + 32 + 32;
+
+        byte _apiVersion = 0;
+        short _peerCount = 0;
+        byte[] _pendingTxCount = new byte[0];
+        int _latency = 0;
+        if (_bytes.length > Math.max(_len, minLenNew)) {
```

Comment:
why just use _bytes.length > minLenNew ? cuase it's new decode logic

Reply:
totalDifficultyLen is not a fixed value. When decoding the old version, i think _bytes.length can be larger than minLenNew if totalDifficultyLen is large.